src/go/rpk/pkg/cli/acl/user/create.go

```diff
@@ -102,17 +95,48 @@ acl help text for more info.
 			err = cl.CreateUser(cmd.Context(), user, pass, mechanism)
 			out.MaybeDie(err, "unable to create user %q: %v", user, err)
 			fmt.Printf("Created user %q.\n", user)
+			if generated {
+				fmt.Println("Automatically generated password:")
+				// In a new line so it can be easily 'tail -1'-ed
+				fmt.Println(pass)
+			}
 		},
 	}
 
 	cmd.Flags().StringVar(&userOld, "new-username", "", "")
 	cmd.Flags().MarkHidden("new-username")
 
-	cmd.Flags().StringVar(&pass, "password", "", "New user's password (NOTE: if using --password for the admin API, use --new-password)")
+	// This is needed here in order to show the password flag with a different
+	// usage text for this command only.
+	p.InstallKafkaFlags(cmd)
+	passwordFlag := cmd.PersistentFlags().Lookup("password")
+	passwordFlag.Usage = "New user's password (NOTE: if using --password for the admin API, use --new-password)"
+	passwordFlag.Hidden = false
+
 	cmd.Flags().StringVarP(&newPass, "new-password", "p", "", "")
 	cmd.Flags().MarkHidden("new-password")
 
 	cmd.Flags().StringVar(&mechanism, "mechanism", strings.ToLower(admin.ScramSha256), "SASL mechanism to use for the user you are creating (scram-sha-256, scram-sha-512, case insensitive)")
 
 	return cmd
 }
+
+func generatePassword(passLength int) (string, error) {
+	lowercase := "abcdefghijklmnopqrstuvwxyz"
+	uppercase := strings.ToUpper(lowercase)
+	numbers := "0123456789"
+	special := ".,&_+|[]/-()"
```

Comment:
wdyt about not having special characters? They're a bit of a pain to use as a flag since a lot of them need to be escaped. We have quite a lot of entropy with 30 alnum characters -- worst case, we could increase to 40 or somethign.

Reply:
I strongly agree with not using special characters in the autogen password. It's fine that RP accepts them but there's no reason to inflict them on people if we don't have to.